lib/opFns.js

```diff
@@ -682,13 +716,75 @@ module.exports = {
       }
     })
   },
+  STATICCALL: function (gasLimit, toAddress, inOffset, inLength, outOffset, outLength, runState, done) {
+    var stateManager = runState.stateManager
+    gasLimit = new BN(gasLimit)
+    toAddress = utils.setLengthLeft(toAddress, 20)
+    var value = new BN(0)
+    inOffset = utils.bufferToInt(inOffset)
+    inLength = utils.bufferToInt(inLength)
+    outOffset = utils.bufferToInt(outOffset)
+    outLength = utils.bufferToInt(outLength)
+
+    var data = memLoad(runState, inOffset, inLength)
+
+    var options = {
+      gasLimit: gasLimit,
+      value: value,
+      to: toAddress,
+      data: data,
+      static: true
+    }
+
+    var localOpts = {
+      inOffset: inOffset,
+      inLength: inLength,
+      outOffset: outOffset,
+      outLength: outLength
+    }
+
+    stateManager.exists(toAddress, function (err, exists) {
+      if (err) {
+        done(err)
+        return
+      }
+
+      stateManager.accountIsEmpty(toAddress, function (err, empty) {
+        if (err) {
+          done(err)
+          return
+        }
+
+        try {
+          checkCallMemCost(runState, options, localOpts)
+          checkOutOfGas(runState, options)
+        } catch (e) {
+          done(e.error)
+          return
+        }
+
+        makeCall(runState, options, localOpts, done)
+      })
+    })
+  },
   RETURN: function (offset, length, runState) {
     offset = utils.bufferToInt(offset)
     length = utils.bufferToInt(length)
     runState.returnValue = memLoad(runState, offset, length)
   },
+  REVERT: function (offset, length, runState) {
+    offset = utils.bufferToInt(offset)
+    length = utils.bufferToInt(length)
+
+    runState.stopped = true
+    runState.returnValue = runState.lastReturned = memLoad(runState, offset, length)
```

Comment:
It is not necessary to set `runState.lastReturned` here.  The return data buffer is already set at https://github.com/jwasinger/ethereumjs-vm/blob/wip/metropolis/lib/opFns.js#L1022